### ISSUE TITLE
Button does not resize to fit text

### ISSUE THREAD

**Issue by im-jersh** [luispadron/UIEmptyState#16]
It seems like setting the font size to some arbitrary value depending on the device type will cause the action button's title to be truncated.

Here's an 8 Plus with the font size set to 17pt:
![simulator screen shot - iphone 8 plus - 2018-02-22 at 14 22 37](https://user-images.githubusercontent.com/4296435/36562763-93fc072a-17dd-11e8-9783-20c987be023e.png)

And the same device with the font size set to 16pt:
![simulator screen shot - iphone 8 plus - 2018-02-22 at 14 23 07](https://user-images.githubusercontent.com/4296435/36562805-b1a72caa-17dd-11e8-8c10-a0a755507732.png)

**Comment by luispadron** [comment]
You need to adjust the buttons size to fit the font, you can do this with the data source methods.

**Comment by im-jersh** [comment]
But even the documentation says the default implementation should size the button to fit the title PLUS some padding yet this is not the result I'm seeing. Don't mind sizing the button myself, just think the docs should be consistent with that requirement if it is in fact a requirement.

![screen shot 2018-02-22 at 3 28 12 pm](https://user-images.githubusercontent.com/4296435/36565147-3432220e-17e5-11e8-94d2-e8c90dcdd8f0.png)

**Comment by im-jersh** [comment]
Also, if simulate padding by adding blank spaces to the beginning and end of my button title, then the issue is resolved. Definitely think this is issue is a bug that needs to be addressed internally. Can try to track it down later and submit a PR.

**Comment by luispadron** [comment]
You're right, my bad. This does seem like a bug that should be looked into.

The code which handles this constraint for the button:

```swift
let size = buttonSize ?? self.buttonTitle?.size() ??
                    self.buttonImage?.size ??
                    CGSize(width: 0, height: 0)
                
button.heightAnchor.constraint(equalToConstant: size.height).isActive = true
button.widthAnchor.constraint(equalToConstant: size.width).isActive = true
```

I don't have time currently to look at this, but if you want to take a look and submit a PR that would be great!
